test/dotnet-sln-add.Tests/GivenDotnetSlnAdd.cs

```diff
@@ -788,6 +853,32 @@ private void WhenSlnContainsSolutionFolderWithDifferentCasingItDoesNotCreateDupl
             solutionFolderProjects.Count().Should().Be(1);
         }
 
+        [Fact]
+        public void WhenProjectIsAddedToSolutionConfigurationsAreMappedCorrectly()
+        {
+            var slnDirectory = TestAssets
+                .Get("TestAppWithSlnAndProjectConfigs")
+                .CreateInstance()
+                .WithSourceFiles()
+                .Root
+                .FullName;
+
+            var slnFullPath = Path.Combine(slnDirectory, "App.sln");
+
+            var result = new DotnetCommand()
+                .WithWorkingDirectory(slnDirectory)
+                .ExecuteWithCapturedOutput($"sln add App");
+            result.Should().Pass();
+
+            result = new DotnetCommand()
+                .WithWorkingDirectory(slnDirectory)
+                .ExecuteWithCapturedOutput($"sln add Lib");
+            result.Should().Pass();
+
+            File.ReadAllText(slnFullPath)
+                .Should().BeVisuallyEquivalentTo(ExpectedSlnFileAfterAddingProjectsWithPlatformsAndConfigs);
```

Comment:
nit pick: This tests is too "integration". As a result, it is hard to understand what is changed, what is the "important" part you want to assert from the test.  For example, is "VisualStudioVersion = 15.0.26006.2" important?

Also, this test will be easy to break due to unrelated change, like VisualStudioVersion move to 16

Reply:
Great feedback, thanks.  It does test two things at once, which is that App should have the configurations all mapped to "AnyCPU" because it is the only platform it supports while Lib should map to all three supported platforms; simultaneously ignoring configurations/platforms not present at the solution level.  I can break this down into discrete tests.

Reply:
With regards to being fragile due to version numbers in the solution file, the existing tests also appear to suffer from this.  Additionally, it would only be an issue if `sln add` modified the solution version when writing out the modified solution file; as far as I know, it doesn't even look at the version numbers.